### ISSUE TITLE
planetarydata.org accessibility scan part 2

### ISSUE THREAD

**Issue by jordanpadams** [NASA-PDS/planetarydata.org#16]

**Comment by nutjob4life** [comment]
@jordanpadams unfortunately my trial period of [Deque's Axe DevTools](https://www.deque.com/axe/devtools/) has expired so I can't use it for accessibility scans anymore.

Can we purchase a copy? If not, is there a resource at JPL we can use? I know the URS folks do a 508 scan of sites before they get a clearance.

**Comment by jordanpadams** [comment]
@anilnatha @edwardbarraza @eddiesarevalo any ideas? ☝️

**Comment by jordanpadams** [comment]
@anilnatha @edwardbarraza @eddiesarevalo ☝️ ???

**Comment by anilnatha** [comment]
I can run the scan using SortSite.

**Comment by jordanpadams** [comment]
Blocked waiting for a tool to do this

**Comment by nutjob4life** [comment]
@anilnatha that'd be great. The site address = https://ipda.jpl.nasa.gov/

**Comment by anilnatha** [comment]
@jordanpadams Do we also need to run this for planetarydata.org as the title states?

**Comment by nutjob4life** [comment]
@anilnatha, let me answer that for @jordanpadams: 

"ipda.jpl.nasa.gov" will become "planetarydata.org" when the site is ready to go live.

The current content at "planetarydata.org" will go away. It doesn't need to be scanned. But thanks for checking! 😁

**Comment by anilnatha** [comment]
Thanks for the clarification @nutjob4life!

**Comment by anilnatha** [comment]
@jordanpadams @nutjob4life The accessibility scan using the SortSite software is complete and I've exported the results to our gdrive => https://drive.google.com/drive/folders/1Hb8A4xPWTObU3jXs9Cvr2VO0tfrKsk65?usp=share_link

We should probably cross reference this with the results from the Axe browser plugin that is available for Chrome/Firefox.

**Comment by jordanpadams** [comment]
thanks @anilnatha

**Comment by tloubrieu-jpl** [comment]
@nutjob4life is solving the issues raised by the scan

**Comment by nutjob4life** [comment]
Colleagues:

Thanks much for the report generously generated by @anilnatha.

For some clarification, this isn't a US Government [Section 508](https://www.section508.gov/) scan. It's a [WCAG](https://www.w3.org/WAI/standards-guidelines/#guidelines) scan, which given the _international_ nature of the site is perhaps more appropriate.

Second, the new IPDA website will neither be a US Government site nor an international `.int` website, so we get to pick how much conformance we want–or not. Please keep that in mind as I point out what's in this report.


### Errors

The scan found several errors:

1. **SSL errors**: After the site was moved from `pdscloud-ipda` to `ipda`, numerous image links broke. This is a WordPress issue, as WordPress doesn't seem to like having its domain changed. I fixed all that I could find. But the scan exposed two additional image links that were still pointing to `pdscloud-ipda`. I've fixed these as well. **However**, once the site then goes to `planetarydata.org`, we should expect image links to break _again_. (This is something other CMS's handle better.)
2. **Broken links**: There was a link to an old `oodt` site on the "Meetings" page. After changing the "Meetings" page the way TS and CA wanted, this problem went away. There's also a broken link to https://cnes.fr/ on the partners page. There's nothing we can do about this: the website for CNES has been down for _weeks_.
3. **Spelling errors**: all false positives except for one: `archivex` → `archive`.


### Accessibility

For level A compliance with WCAG, we have to address the following:

1. Links on the Minutes page use the exact same text but go to different destinations. This has been fixed.
2. `aria-labelledby` references a blank element on the "Access to Data" page. This attribute not longer appears on the page.
3. The element `footer` must not appear as a descendant of `footer`. I cannot fix this, but we can report it to the theme developer.
4. The element `header` must not appear as a descendant of `header`. Ditto.

For AA compliance, there are several other issues we can address, mostly due to finicky issues with "if a color is set on one element it must be set on all elements". I don't think it's necessary to pursue this.

For AAA compliance, there are no other issues that need to be addressed.


### Compatibility

Priority 1 compatibility has to do with some IE11 issues. I recommend we ignore these as Edge is now the standard Microsoft browser (and Edge is just Chrome in Microsoft regalia).

Priority 2 compatibility is yet more finicky IE11 issues. Ditto on ignoring these.

Priority 3 compatibility affects IE and Safari on iPhone and iPad and support of `margin-block-` and `margin-inline` properties, as well as some other niche CSS properties.

This is getting quite fussy, but as a safety-net I used Safari on macOS in Responsive Design Mode to browse the site as an iPhone SE, iPhone 8 Plus, iPad 7.9″, and iPad Pro 12.9″. Everything looked fine to me. 


### Search

We've already hit Priority 1 and Priority 2. To reach Priority 3 we'd need to add a `meta` description to every page. The "Yoast" plug-in could do this. Let me know if you want it installed.


### Standards

Priority 1 issues with `header` contained in `header` and `footer` contained in `footer` are duplicated here. There is apparently a CSS validation error but I believe it's a false positive.  There is a `style` element in the footer that should be in the `head` section and I have no idea where it's come from; this should also be reported to the theme provider as a bug.

Allegedly the `title` element is empty on the "Access to Data" page, but this is a false positive.

There's a `@charset UTF-8` in the "Partners" page that doesn't belong, but it's harmless as the entire site is already UTF-8.

There's an invalid CSS selector in various places that is also harmless—in addition to some other CSS identifiers, and some CSS values. All harmless.

There's an invalid CSS `url()` in two places but the line numbers indicated are not found in the resulting HTML output.

I don't think it's worth anyone's time to try to address the Priority 2 or Priority 3 issues. If they absolutely must be resolved, then it'd be best to select a different theme altogether—or a different CMS—one that has WCAG AAA compliance.


### Usability

- Underlined text: false positive.
- Keep URLs shorter than 78 characters: changed the slug for the "Archiving Components for International MOUs" to be much smaller and adjusted the link from the Recommendations & Standards page as well as on the header navigation.
- Provide a search page. True, we don't have a search box. Do we want one?

**Comment by jordanpadams** [comment]
@nutjob4life thanks for the awesomely thorough scrub of this. from your report, it sounds like we are good. I don't think we need to move forward with any of this, but kind of crazy these free themes are not accessible out of the box. but whatever.

> Provide a search page. True, we don't have a search box. Do we want one?

Nah. There is really no content on the site so doesn't seem to make a lot of sense.

in the end, I think we are go for launch 🚀 . URS submitted and now we hurry up and wait

**Comment by nutjob4life** [comment]
@jordanpadams 
![Whew!](https://user-images.githubusercontent.com/814813/224379364-2e39be85-019e-4629-9c18-ac4cfd4330f8.gif)
